worlds/pokemon_rb/__init__.py

```diff
@@ -467,13 +467,15 @@ def number_of_zones(mon):
         locs = {self.multiworld.get_location("Fossil - Choice A", self.player),
                 self.multiworld.get_location("Fossil - Choice B", self.player)}
 
-        for loc in locs:
+        if not self.multiworld.key_items_only[self.player]:
             if self.multiworld.fossil_check_item_types[self.player] == "key_items":
```

Comment:
```suggestion
        if not self.multiworld.key_items_only[self.player]:
            rule = None
            if self.multiworld.fossil_check_item_types[self.player] == "key_items":
```

Need to establish rule before using it. None isn't necessarily the correct thing to choose here though, I'm just not super sure what is without reviewing the code further.

Reply:
Another option would be to actually have an "else:" case, in which case declaring it before becomes unnecessary, although it'd be effectively the same thing

Reply:
Ahh right, I forgot python is nice like that, letting you have it if every case leads you to setting rule as something